app/jobs/jobs/calls_monitoring.rb

```diff
@@ -289,13 +296,28 @@ def detect_customers_auth_calls_to_reject
     # detect gateway is disabled by orig_gw_id and term_gw_id
     def detect_gateway_calls_to_reject
       flatten_calls.each do |call|
-        if disabled_orig_gw_active_calls.key?(call[:orig_gw_id]) || disabled_term_gw_active_calls.key?(call[:term_gw_id])
-          local_tag = call[:local_tag]
-          @terminate_calls[local_tag] = call
+        if disabled_orig_gw_active_calls.key?(call[:orig_gw_id])
+          terminate_calls_for_disabled_orig_gw(call)
+        elsif disabled_term_gw_active_calls.key?(call[:term_gw_id])
+          terminate_calls_for_disabled_term_gw(call)
         end
       end
     end
```

Comment:
I think better would be to split `detect_gateway_calls_to_reject` into 2 methods: `detect_orig_gateway_calls_to_reject` and `detect_term_gateway_calls_to_reject` and use same approach that was used in `detect_customers_auth_calls_to_reject`

**P.S.** also please refactor adding calls to `@terminate_calls`, they all should use `terminate_calls` method

```ruby
    def detect_orig_gateway_calls_to_reject
      return unless teardown_enabled?(:teardown_on_disabled_orig_gw)

      calls_to_terminate = flatten_calls.select do |call|
        disabled_orig_gw_active_calls.key?(call[:orig_gw_id])
      end
      terminate_calls(calls_to_terminate)
    end

    def detect_term_gateway_calls_to_reject
      return unless teardown_enabled?(:teardown_on_disabled_term_gw)

      # same
    end

    def terminate_calls(calls)
      @terminate_calls.merge!(calls.index_by { |c| c[:local_tag] })
    end
```